### DISCUSSION TITLE
Esp8266 stuck with HC-SR04 that working in uart automatic mode (sending data to every 100ms)

### DISCUSSION THREAD

**Discussion by midnight1979** [arendst/Tasmota#15976] [Q&A] [answered]
### PROBLEM DESCRIPTION
_A clear and concise description of what the problem is._

HC-SR04 in uart mode, automatically sending range in cm to esp8266.
Esp8266 hangs after different period of working. It may work 5 minutes, and may work several hours, some times in web log having messages "CRC error"...
But after esp8266 hangs, only reset button helps to bring it to "life".
I have 2 another devices with hc-sr04 that working in trig/echo mode and it is OK. But with the sensor in automatic uart send mode - having these troubles. 

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): esp8266 wemos d1
- [ ] Tasmota binary firmware version number used: 12
  - [x] Pre-compiled
  - [ ] Self-compiled
- [x] Flashing tools used: tasmotizer
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:


```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:


```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior:_

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by sfromis** [comment, marked as answer]
Well, the Tasmota support for SR04 is with trig/echo pins..... Outside of that, usage is basically experimental.

**Comment by midnight1979** [comment]
combination esp8266+hc-sr04 (uart)+tasmota is working but not stable... maybe dev can find the solution....
Link for sensor that I'm using:
https://a.aliexpress.com/_DkDsYjp
